Common subdirectories: a/.git and b/.git
diff -u a/config.h b/config.h
--- a/config.h	2020-01-24 10:57:59.364459732 -0500
+++ b/config.h	2020-02-25 11:15:42.136147583 -0500
@@ -3,34 +3,45 @@
 
 #define MOD Mod4Mask
 
+/* layout(s) */
+static int mas = 60;    /* master area size fudge */
+static int gap = 20;	/* gap around windows */
+
+
 const char* menu[]    = {"dmenu_run",      0};
 const char* term[]    = {"st",             0};
 const char* scrot[]   = {"scr",            0};
-const char* briup[]   = {"bri", "10", "+", 0};
-const char* bridown[] = {"bri", "10", "-", 0};
 const char* voldown[] = {"amixer", "sset", "Master", "5%-",         0};
 const char* volup[]   = {"amixer", "sset", "Master", "5%+",         0};
 const char* volmute[] = {"amixer", "sset", "Master", "toggle",      0};
-const char* colors[]  = {"bud", "/home/goldie/Pictures/Wallpapers", 0};
+const char* colors[]  = {"bud", "/home/chef/Pictures/Wallpapers", 0};
+const char* surfcmd[] = {"firefox", "https://news.google.com", 0};
+const char* editcmd[] = {"geany", 0};
+const char* somacmd[] = {"dmenu_soma.sh", 0};
 
 static struct key keys[] = {
     {MOD,      XK_q,   win_kill,   {0}},
     {MOD,      XK_c,   win_center, {0}},
     {MOD,      XK_f,   win_fs,     {0}},
+    {MOD,      XK_n,   win_l_fifty,{0}},
+    {MOD,      XK_m,   win_r_fifty,{0}},
+    {MOD,      XK_t,   win_t_fifty,{0}},
+    {MOD,      XK_b,   win_b_fifty,{0}},
 
     {Mod1Mask,           XK_Tab, win_next,   {0}},
     {Mod1Mask|ShiftMask, XK_Tab, win_prev,   {0}},
 
-    {MOD, XK_d,      run, {.com = menu}},
+    {Mod1Mask|ShiftMask, XK_d, run, {.com = menu}},
     {MOD, XK_w,      run, {.com = colors}},
     {MOD, XK_p,      run, {.com = scrot}},
     {MOD, XK_Return, run, {.com = term}},
 
-    {0,   XF86XK_AudioLowerVolume,  run, {.com = voldown}},
-    {0,   XF86XK_AudioRaiseVolume,  run, {.com = volup}},
-    {0,   XF86XK_AudioMute,         run, {.com = volmute}},
-    {0,   XF86XK_MonBrightnessUp,   run, {.com = briup}},
-    {0,   XF86XK_MonBrightnessDown, run, {.com = bridown}},
+    {MOD, XK_F10,    run, {.com = voldown}},
+    {MOD, XK_F11,    run, {.com = volup}},
+    {MOD, XK_F12,    run, {.com = volmute}},
+    {Mod1Mask|ShiftMask, XK_f, run,   {.com = surfcmd}},
+    {Mod1Mask|ShiftMask, XK_g, run,   {.com = editcmd}},
+    {Mod1Mask|ShiftMask, XK_r, run,   {.com = somacmd}},
 
     {MOD,           XK_1, ws_go,     {.i = 1}},
     {MOD|ShiftMask, XK_1, win_to_ws, {.i = 1}},
Only in b/: sowm
diff -u a/sowm.c b/sowm.c
--- a/sowm.c	2020-02-22 08:51:48.095431276 -0500
+++ b/sowm.c	2020-02-25 11:17:55.367137355 -0500
@@ -132,6 +132,30 @@
     XMoveWindow(d, cur->w, (sw - ww) / 2, (sh - wh) / 2);
 }
 
+void win_l_fifty(const Arg arg) {
+    if (!cur) return;
+
+    XMoveResizeWindow(d, cur->w, 0 +gap, 0 +gap, sw/2 -gap +mas, sh - 2*gap);
+}
+
+void win_r_fifty(const Arg arg) {
+    if (!cur) return;
+
+    XMoveResizeWindow(d, cur->w, sw/2 +gap + mas, 0 +gap, sw/2 - 2*gap -mas, sh -2*gap);
+}
+
+void win_t_fifty(const Arg arg) {
+    if (!cur) return;
+
+    XMoveResizeWindow(d, cur->w, sw/2 +gap +mas, 0 +gap, sw/2 - 2*gap -mas, sh/2 -gap);
+}
+
+void win_b_fifty(const Arg arg) {
+    if (!cur) return;
+
+    XMoveResizeWindow(d, cur->w, sw/2 +gap +mas, sh/2 +gap, sw/2 - 2*gap -mas, sh/2 - 2*gap);
+}
+
 void win_fs(const Arg arg) {
     if (!cur) return;
 
@@ -141,7 +165,7 @@
 
     } else {
         XMoveResizeWindow(d, cur->w, cur->wx, cur->wy, cur->ww, cur->wh);
-    }
+	}
 }
 
 void win_to_ws(const Arg arg) {
diff -u a/sowm.h b/sowm.h
--- a/sowm.h	2020-02-22 08:51:48.095431276 -0500
+++ b/sowm.h	2020-02-22 09:47:48.568173291 -0500
@@ -45,6 +45,10 @@
 void run(const Arg arg);
 void win_add(Window w);
 void win_center(const Arg arg);
+void win_l_fifty(const Arg arg);
+void win_r_fifty(const Arg arg);
+void win_t_fifty(const Arg arg);
+void win_b_fifty(const Arg arg);
 void win_del(Window w);
 void win_fs(const Arg arg);
 void win_focus(client *c);
Only in b/: sowm.o
